drivers/gpio/gpio_ite_it8xxx2.c

```diff
@@ -233,6 +349,8 @@ static int gpio_ite_configure(const struct device *dev,
 	volatile uint8_t *reg_gpdr = (uint8_t *)gpio_config->reg_gpdr;
 	volatile uint8_t *reg_gpcr = (uint8_t *)(gpio_config->reg_gpcr + pin);
 	volatile uint8_t *reg_gpotr = (uint8_t *)gpio_config->reg_gpotr;
+	volatile uint8_t *reg_1p8v = gpio_1p8v[gpio_config->idx][pin].reg_1p8v;
```

Comment:
Let's add an __ASSERT / __ASSERT_NO_MSG which verifies that `gpio_config->idx` and `pin` is withing limits.

Reply:
The __ASSERT has to be called before the variable is used as an index.

I've made one mistake. The value of the `pin` argument is verified in `include/drivers/gpio.h` so there is no need to do it here again. Only gpio_config->idx needs to be verified (unless you're able to remove it completely as suggested by @galak).

Reply:
Done. (change to verify gpio_config->index only).

Reply:
Thanks, however you still need to make sure that __ASSERT is called before the variable is used as an index. I.e. before the line with `gpio_1p8v[gpio_config->index][pin].offset`. If index value is invalid you're going to access out of bound array element.

Reply:
Done.